### PR TITLE
Molstar volseg extension with custom viewer

### PR DESCRIPTION
**Pull request by aliaksei-chareshneu**

### CODE REVIEW

File: app.ts [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+    'mp4-export': PluginSpec.Behavior(Mp4Export),
+    'geo-export': PluginSpec.Behavior(GeometryExport),
+    'ma-quality-assessment': PluginSpec.Behavior(MAQualityAssessment),
+    'zenodo-import': PluginSpec.Behavior(ZenodoImport),
+    'sb-ncbr-partial-charges': PluginSpec.Behavior(SbNcbrPartialCharges),
+    'wwpdb-chemical-component-dictionary': PluginSpec.Behavior(wwPDBChemicalComponentDictionary),
+    'mvs': PluginSpec.Behavior(MolViewSpec),
+    'cvsx': PluginSpec.Behavior(CVSXSpec),
+    'tunnels': PluginSpec.Behavior(SbNcbrTunnels),
+};
```
Comment (dsehnal) [inline app.ts:92]: Let's only include the volseg and cvsx extension here.

File: app.ts [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+        await this.plugin.builders.structure.hierarchy.applyPreset(trajectory, 'all-models', { useDefaultIfSingleModel: true, representationPresetParams: options?.representationParams });
+    }
+
+    async loadStructureFromData(data: string | number[], format: BuiltInTrajectoryFormat, options?: { dataLabel?: string }) {
+        const _data = await this.plugin.builders.data.rawData({ data, label: options?.dataLabel });
+        const trajectory = await this.plugin.builders.structure.parseTrajectory(_data, format);
+        await this.plugin.builders.structure.hierarchy.applyPreset(trajectory, 'default');
+    }
+
+    loadPdb(pdb: string, options?: LoadStructureOptions) {
```
Comment (dsehnal) [inline app.ts:270]: I don't think all of these helper functions are needed, lets make this focused on on the volseg extension

File: index.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<meta http-equiv="refresh" content="0; url=https://raw.githubusercontent.com/molstar/molstar-volseg/molstar-volseg-extension/molstar-extension/site/index.html">
```
Comment (dsehnal) [inline index.html:1]: Why is this page here?

File: molstar-extension/src/apps/viewer/index.html [hunk 4]
```diff
@@ -0,0 +1,131 @@
+<!DOCTYPE html>
```
Comment (dsehnal) [inline molstar-extension/src/apps/viewer/index.html:1]: Change the parent folder to just `src/viewer`

File: molstar-extension/src/volumes-and-segmentations/entry-geometric-segmentation.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
+/**
```
Comment (dsehnal) [inline molstar-extension/src/volumes-and-segmentations/entry-geometric-segmentation.ts:1]: Move `csv-extension`, `meshes`, `volumes-and-segmentations` to `extension/*` folder.

File: molstar-extension/site/molstar.mjs
Comment (dsehnal) [inline molstar-extension/site/molstar.mjs:1]: Why is this file here?

File: app.ts [hunk 6]
```diff
@@ -0,0 +1,601 @@
+/**
```
Comment (dsehnal) [inline app.ts:1]: Why is this file here?

File: molstar-extension/esbuild.config.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  // external: ['/images/*'],
+  loader: {
+    '.png': 'file',
+    '.jpg': 'file',
+    '.svg': 'dataurl'
+  },
+  logLevel: productionMode ? 'error' : 'info',
+  minify: true,
+  sourcemap: !productionMode && 'linked',
+  plugins: [sassPlugin()],
```
Comment (dsehnal) [inline molstar-extension/esbuild.config.js:24]: Hmm, you actually do have it here. No need have `molstar.css` in site them.

File: molstar-extension/site/index.html [hunk 8]
```diff
@@ -0,0 +1,131 @@
+<!DOCTYPE html>
```
Comment (dsehnal) [inline molstar-extension/site/index.html:1]: Is this the built version of the site? Should we do this just in CI perhaps?

File: index.html (hunk 3 above)
Comment (aliaksei-chareshneu) [reply index.html:1]: @dsehnal, initially I tried to deploy it from github UI, which allows to deploy gh pages only from root or docs

File: molstar-extension/site/index.html (hunk 8 above)
Comment (aliaksei-chareshneu) [reply molstar-extension/site/index.html:1]: Yes, sure

File: molstar-extension/src/volumes-and-segmentations/entry-geometric-segmentation.ts (hunk 5 above)
Comment (aliaksei-chareshneu) [reply molstar-extension/src/volumes-and-segmentations/entry-geometric-segmentation.ts:1]: @dsehnal, you mean directly in `molstar-extension`?

File: molstar-extension/esbuild.config.js (hunk 7 above)
Comment (aliaksei-chareshneu) [reply molstar-extension/esbuild.config.js:24]: @dsehnal, you mean that I can use `main.css`
```
@import "../../node_modules/molstar/lib/mol-plugin-ui/skin/light.scss";
@import "../../node_modules/jsoneditor/dist/jsoneditor.css";
@import "../../node_modules/reactjs-popup/dist/index.css";
```


directly in `index.html` and do not need to compile them in `molstar.css`?

File: molstar-extension/src/volumes-and-segmentations/entry-geometric-segmentation.ts (hunk 5 above)
Comment (dsehnal) [reply molstar-extension/src/volumes-and-segmentations/entry-geometric-segmentation.ts]: - `molstar-extension/src/extension/*` will have all the extension-related code
- `molstar-extension/src/viewer/*` will have the viewer

File: molstar-extension/esbuild.config.js (hunk 7 above)
Comment (dsehnal) [reply molstar-extension/esbuild.config.js:24]: Actaully, no. Should change this import `require('mol-plugin-ui/skin/light.scss');` .... I guess this should be `molstar/lib/...`, no?

File: molstar-extension/src/volumes-and-segmentations/entry-geometric-segmentation.ts (hunk 5 above)
Comment (aliaksei-chareshneu) [reply molstar-extension/src/volumes-and-segmentations/entry-geometric-segmentation.ts]: Ok, thanks
